### PR TITLE
feat: implement persons

### PR DESCRIPTION
**Pull request by idax6797**
> [!NOTE]  
> I have set up self-linking made a Link class for this and referred that in the controller.
> I have to implement the linking to the media when that method is done. 

**This pull request introduces a new feature to manage and retrieve information about persons.**

 It includes the addition of a new `PersonController` to handle API requests, the creation of new data transfer objects, and the implementation of repository and manager classes for persons and their related data. 

### New Controller:
* [`src/CitMovie.Api/Controller/PersonController.cs`](diffhunk://#diff-e3304a4869c9572fc75559e86d4bc92896feb975132c980a2d2bfc75b5ea8feaR1-R176): Added `PersonController` to handle API requests related to persons, including getting persons, media by person ID, and frequent co-actors.

### Data Context Updates:
* [`src/CitMovie.Data/DataContext.cs`](diffhunk://#diff-a62bfdeeb6f78d87dae6da6a3e3fe07bb85addcb56ef0d65dabf85163aa23abfR31): Added `DbSet<CoActor>` and configured many-to-many relationships for `Media` and `Genre`. [[1]](diffhunk://#diff-a62bfdeeb6f78d87dae6da6a3e3fe07bb85addcb56ef0d65dabf85163aa23abfR31) [[2]](diffhunk://#diff-a62bfdeeb6f78d87dae6da6a3e3fe07bb85addcb56ef0d65dabf85163aa23abfR58-R66)

### New Repository and Manager Classes:
* [`src/CitMovie.Data/Repositories/PersonRepository.cs`](diffhunk://#diff-beb193273f6f4e74ab542afbbb3d4e7c5c7976d55f4e04d2afb85a425df20b39R1-R111): Implemented `PersonRepository` to interact with the database for person-related queries.
* [`src/CitMovie.Business/Managers/PersonManager.cs`](diffhunk://#diff-b5dccb6016399030b9d971d3a646e49d85a387659f873b486bb1cf4d81869c9dR1-R71): Implemented `PersonManager` to handle business logic for person-related operations.
* [`src/CitMovie.Business/Managers/IPersonManager.cs`](diffhunk://#diff-85adec54013f31e3acb91fb65f9d5fe7d39e8248b790f1554ed687c6eb8cb4caR1-R15): Defined `IPersonManager` interface.
* [`src/CitMovie.Data/Repositories/IPersonRepository.cs`](diffhunk://#diff-068d7ad2d90f6f216666a6f73deee951536ed3e4db3c6cb6ad3d6265fece3f84R1-R14): Defined `IPersonRepository` interface.

### Data Transfer Objects:
* [`src/CitMovie.Models/DataTransferObjects/PersonDto.cs`](diffhunk://#diff-b38433991e5070497d64d51ea75fbecea152dca5118587f9ad7dd3a503306006R1-R33): Added `PersonResult`, `MediaResult`, and `CoActorResult` DTOs.
* [`src/CitMovie.Models/DomainObjects/CoActors.cs`](diffhunk://#diff-ac92e69863f6968daf992a40cd0b85801f33c8f80d4646f8ecad5e5a169e1a8dR1-R13): Added `CoActor` domain object.
* [`src/CitMovie.Models/Link.cs`](diffhunk://#diff-0662300a96afc498e5e64f27a7d1e44c38e28c35ed58323ca6a81f362d2ed659R1-R8): Added `Link` class for HATEOAS implementation.

### AutoMapper Configuration:
* [`src/CitMovie.Business/AutoMappers/AutoMapperProfile.cs`](diffhunk://#diff-ddcc1c501da289521b934bce8a9907b5229351d49e87e694f155d63262388b2fR29-R40): Configured mappings for `Person`, `Media`, and `CoActor` to their respective DTOs.

### Service Collection Extensions:
* [`src/CitMovie.Business/CitMovieServiceCollectionExtensions.cs`](diffhunk://#diff-dabfe6c1fef4740e18f2895f06dcf816b3560336c995de957529a0af40a905c1R24): Added scoped services for `IPersonRepository` and `IPersonManager`. [[1]](diffhunk://#diff-dabfe6c1fef4740e18f2895f06dcf816b3560336c995de957529a0af40a905c1R24) [[2]](diffhunk://#diff-dabfe6c1fef4740e18f2895f06dcf816b3560336c995de957529a0af40a905c1R35)

### Test Configuration:
* [`src/CitMovie.Business.Tests/PersonManagerUnitTest.cs`](diffhunk://#diff-ddcc1c501da289521b934bce8a9907b5229351d49e87e694f155d63262388b2fR29-R40): Configured tests  for [`src/CitMovie.Business/Managers/PersonManager.cs`](diffhunk://#diff-b5dccb6016399030b9d971d3a646e49d85a387659f873b486bb1cf4d81869c9dR1-R71):  to insure the class interacts correctly with its dependencies and produces the expected results.

### CODE REVIEW

**Review by sofushn — CHANGES_REQUESTED** [review]
.

File: src/CitMovie.Models/DataTransferObjects/PersonDto.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public int MediaId { get; set; }
+    public string Title { get; set; }
+    public string? Type { get; set; }
+    public IEnumerable<string> Genres { get; set; }
+    public string? Description { get; set; }
+    public string? Awards { get; set; }
+    public int BoxOffice { get; set; }
+    public List<Link> Links { get; set; } = new List<Link>();
+}
```
[the remark this answers is not in the document]
Comment (idax6797) [reply src/CitMovie.Models/DataTransferObjects/PersonDto.cs:25]: Yes, we should remember this, i make a task for it.

File: src/CitMovie.Api/Controller/PersonController.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        if (personId.HasValue)
+        {
+            coActor.Links.Add(new Link
+            {
+                Href = HttpContext != null ? _linkGenerator.GetUriByName(HttpContext, nameof(GetPersonById), new { id = personId }) : string.Empty,
+                Rel = "self",
+                Method = "GET"
+            });
+        }
+    }
```
[the remark this answers is not in the document]
Comment (idax6797) [reply src/CitMovie.Api/Controller/PersonController.cs]: Yes this is fixed ✅

File: src/CitMovie.Models/DataTransferObjects/PersonDto.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sofushn) [reply src/CitMovie.Models/DataTransferObjects/PersonDto.cs:25]: If you make this a sub class (class within a class) of the PersonResult it wont break with the duplicate class from media

**Review by sofushn — APPROVED** [review]

**Pull request merged**